lib/src/utils/crypto/native.dart

```diff
@@ -65,6 +78,127 @@ abstract class Cipher {
+  }) async {
+    final algorithm = getAlg(key.length * 8);
+
+    final memSize = sizeOf<IntPtr>() + key.length + initialVector.length + maxHashSize;
+    final memNeeded = malloc.call<Uint8>(memSize);
+    final intPointer = memNeeded.cast<IntPtr>();
+    final keyPointer = memNeeded.elementAt(sizeOf<IntPtr>());
+    final initialVectorPointer = keyPointer.elementAt(key.length);
+
+    final hashValuePointer = initialVectorPointer.elementAt(maxHashSize);
+    final digestAlgo = messageDigestAlgorithm.toNativeUtf8();
+
+    IOSink? outIoSink;
+    final cipherContext = EVP_CIPHER_CTX_new();
+    final mDHashContext = EVP_MD_CTX_new();
+    final digestName = EVP_get_digestbyname(digestAlgo);
+    try {
+      outIoSink = outputFile.openWrite();
+      keyPointer.asTypedList(key.length).setAll(0, key);
+      initialVectorPointer.asTypedList(initialVector.length).setAll(0, initialVector);
+      
+      EVP_EncryptInit_ex(cipherContext, algorithm, nullptr, keyPointer, initialVectorPointer);
+
+      EVP_DigestInit_ex(mDHashContext, digestName, nullptr);
+
+      await inputStream.forEach((bytes) {
+        final memData = malloc.call<Uint8>(sizeOf<IntPtr>() + bytes.length);
+        final dataPointer = memData.elementAt(sizeOf<IntPtr>());
+        dataPointer.asTypedList(bytes.length).setAll(0, bytes);
+        EVP_EncryptUpdate(cipherContext, dataPointer, intPointer, dataPointer, bytes.length);
+        outIoSink!.add(Uint8List.fromList(dataPointer.asTypedList(bytes.length)));
+        EVP_DigestUpdate(mDHashContext, dataPointer, bytes.length);
+        malloc.free(memData);
+      });
+
+      EVP_DigestFinal_ex(mDHashContext, hashValuePointer);
+      final hashBase64Encoded = base64Url.encode(Uint8List.fromList(hashValuePointer.asTypedList(maxHashSize))).replaceAll('=', '');
+
+      return EncryptedFileInfo(
+        key: EncryptedFileKey(
+          algorithrm: algorithmName,
+          key: base64Url.encode(key).replaceAll('=', ''),
+          extractable: true,
+          keyOperations: [KeyOperation.encrypt, KeyOperation.decrypt],
+          keyType: keyType,
+        ),
+        version: version,
+        initialVector: base64.encode(initialVector).replaceAll('=', ''),
+        hashes: {
+          messageDigestAlgorithm: hashBase64Encoded,  
+        }
+      );
+    } catch (e) {
+      throw Exception(e);
+    } finally {
+      malloc.free(memNeeded);
+      EVP_MD_CTX_free(mDHashContext);
+      EVP_CIPHER_CTX_free(cipherContext);
+      await outIoSink?.close();
+    }
+  }
+
+  Future<bool> decryptStream({
+    required Stream<List<int>> inputStream,
+    required File outputFile,
+    required EncryptedFileInfo encryptedFileInfo,
+  }) async {
+    final keyDecoded = base64decodeUnpadded(base64.normalize(encryptedFileInfo.key.key));
+    final initialVectorDecoded = base64decodeUnpadded(base64.normalize(encryptedFileInfo.initialVector));
+    final algorithm = getAlg(keyDecoded.length * 8);
```

Comment:
base on the length of the key, we determine the algorithm to use. if length of key is 16 then, use AES 128 CTR, if 32 then use AES 256 CTR. otherwise return an error.